src/enzo/New_Grid_ReadGrid.C

```diff
@@ -194,8 +197,18 @@ int grid::Group_ReadGrid(FILE *fptr, int GridID, HDF5_hid_t file_id,
     if (fscanf(fptr, "NumberOfParticles = %"ISYM"\n", &NumberOfParticles) != 1) {
             ENZO_FAIL("error reading NumberOfParticles.");
     }
- 
-    if (NumberOfParticles > 0) {
+
+    if (fscanf(fptr, "NumberOfActiveParticles = %"ISYM"\n", &NumberOfActiveParticles) != 1) {
+      // Ugly hack to support restart files from before active particle support
+      NumberOfActiveParticles = 0;
+    }
+
+    // We read these in but don't use them (yet?). Therefore we won't error
```

Comment:
Are there still plans to use this?

Reply:
I've had issues with this part of the code alright. I think (from memory) I need to comment these lines out for restarting when no active particles exist. This probably does require an additional PR though. It would be make sense to raise an issue around these lines.

Reply:
These lines are written out for analysis (e.g. yt) but aren't needed for a restart, so we skip them.
```C++
fprintf(fptr, "PresentParticleTypes = ");
    if (NumberOfParticles)
      fprintf(fptr, "DarkMatter ");
    for (int i = 0; i<EnabledActiveParticlesCount; i++){
      if (ActiveParticleTypeCount[i] > 0) {
        fprintf(fptr, "%s ", EnabledActiveParticles[i]->particle_name.c_str());
      }
    }
    fprintf(fptr, "\n");
    // And their counts
    fprintf(fptr, "ParticleTypeCounts = ");
    if (NumberOfParticles)
      fprintf(fptr, "%"ISYM" ", NumberOfParticles);
    for (int i = 0; i<EnabledActiveParticlesCount; i++){
      if (ActiveParticleTypeCount[i] > 0) {
        fprintf(fptr, "%"ISYM" ", ActiveParticleTypeCount[i]);
      }
    }
    fprintf(fptr, "\n");
```